### PR TITLE
COMMERCE-10059

### PR DESCRIPTION
**Pull request by MahmoudAzzam**

### CODE REVIEW

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by victoroscuro** [comment]
ci:test:commerce

File: modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/model/listener/CPDefinitionModelListener.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+				Indexer<CSDiagramEntry> indexer =
+					IndexerRegistryUtil.nullSafeGetIndexer(
+						CSDiagramEntry.class);
+
+				indexer.reindex(csDiagramEntry);
+			}
+		}
+		catch (SearchException searchException) {
+			throw new ModelListenerException(searchException);
+		}
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/model/listener/CPDefinitionModelListener.java:62]: `public void onAfterUpdate(
		CPDefinition originalCPDefinition, CPDefinition cpDefinition) {

		Indexer<CSDiagramEntry> indexer =
			IndexerRegistryUtil.nullSafeGetIndexer(
				CSDiagramEntry.class);

		try {
			indexer.reindex(_csDiagramEntryLocalService.
				getCPDefinitionRelatedCSDiagramEntries(
					cpDefinition.getCPDefinitionId()));
		}
		catch (SearchException searchException) {
			throw new ModelListenerException(searchException);
		}`

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/model/listener/CPDefinitionModelListener.java:62]: @MahmoudAzzam I think you don't need to have a new reference to the indexer for each diagram entries, you should like the suggestion I've put here.

Thanks!

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
@MahmoudAzzam please remember to follow the Workflow we explained it lots of time, if you still have doubts you should ask for clarification.

Thanks!

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/model/listener/CPDefinitionModelListener.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+			List<CSDiagramEntry> csDiagramEntries =
+				_csDiagramEntryLocalService.
+					getCPDefinitionRelatedCSDiagramEntries(
+						cpDefinition.getCPDefinitionId());
+
+			Indexer<CSDiagramEntry> indexer =
+				IndexerRegistryUtil.nullSafeGetIndexer(CSDiagramEntry.class);
+
+			for (CSDiagramEntry csDiagramEntry : csDiagramEntries) {
+				indexer.reindex(csDiagramEntry);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/model/listener/CPDefinitionModelListener.java:56]: why again the cycle? I've put the solution code in the comment last time I reviewed this, you can pass the collection directly inlining the call to the service

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/model/listener/CPDefinitionModelListener.java (hunk 2 above)
Comment (MahmoudAzzam) [reply modules/apps/commerce/commerce-shop-by-diagram-service/src/main/java/com/liferay/commerce/shop/by/diagram/internal/model/listener/CPDefinitionModelListener.java]: Sorry, I completely overlooked it

**Comment by MahmoudAzzam** [comment]
ci:test:sf

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Review by andrea-ale-sbarra — APPROVED** [review]

**Comment by andrea-ale-sbarra** [comment]
Thank you @MahmoudAzzam , please always put the link to the jira ticket in the PR first commit.

**Comment by 22LucaGiannotta22** [comment]
ci:test:commerce

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by 22LucaGiannotta22** [comment]
ci:forward

**Pull request closed without merging**